files/biscotti/app/extensions/authorize_dsl.rb

```diff
@@ -23,7 +23,7 @@ def handle_authorize!
 
         def authorize!
           halt(400) unless payload.key?("token")
-          halt(401) unless valid_token?(payload["token"])
+          halt(401) if payload["token"] != "" && !valid_token?(payload["token"])
```

Comment:
I wonder if we should make `valid_token?()` return false if the string is empty rather than introducing another case that contradicts what the token checking method name implies.

Reply:
@ryancragun The core problem is that it does return false as `token = ""` and `hashed_token` has a value.

Reply:
Thinking about your suggestion more, we can just do the inverse and return `true` if the string is empty.

Reply:
I was thinking `halt(401) unless valid_token?(payload["token"])` where `valid_token?()` when given an empty string returns false.